File: leetcode/0412-fizz-buzz/fizzBuzz.go
package leetcode

import "strconv"

func fizzBuzz(n int) []string {
	var result = []string{}

	for i := 1; i <= n; i++ {
		var item = ""

		if i%3 == 0 {
			item = "Fizz"
		}

		if i%5 == 0 {
			item = item + "Buzz"
		}

		if item == "" {
			item = strconv.Itoa(i)
		}

		result = append(result, item)
	}

	return result
}

File: leetcode/0412-fizz-buzz/fizzBuzz.ts
function fizzBuzz(n: number): string[] {
  const result: string[] = [];

  for (let index = 1; index <= n; index++) {
    let item = "";

    if (index % 3 == 0) {
      item = "Fizz";
    }

    if (index % 5 == 0) {
      item = `${item}Buzz`;
    }

    if (item === "") {
      item = `${index}`;
    }

    result.push(item);
  }

  return result;
}

export default fizzBuzz;
